src/assets/DockerCompose/docker_compose.md

```diff
@@ -0,0 +1,210 @@
+---
+title: Containerize a Web Application using docker compose
+authorName: Sagar Gurung
+authorAvatar: https://avatars.githubusercontent.com/u/46086950?v=4
+authorLink: https://github.com/SagarGi
+createdAt: Nov 27, 2023
+tags: docker-compose, docker, containerization, yaml, services
+banner: https://blog.jankaritech.com/src/assets/DockerCompose/images/docker-cmopose-background.png
+---
+
+In this blog we will basically be containerizing a web application consisting of `MySQL`, `phpmyadmin`, `frontend` and `backend` as services using docker compose.
+Also, we briefly get into how all those services are running and how multiple services make up an application running in multiple containers.
+> Note: This article does not provide concepts of docker. To know concepts of docker [visit here.](https://blog.jankaritech.com/#/blog/How%20to%20use%20selenium%20in%20docker%3F)
+
+## Brief About Docker Compose
+In short, docker compose helps to define and run docker applications in multiple containers. With docker compose we can define a `yml` file where we can define our services, networks, and the volumes required for our application to get it running. Basically, configuring, defining and managing all the necessary things required for our application becomes easy with docker compose.
+
+And in this blog we will also be defining and configuring simple services for our application through a compose file and run it using docker compose.
+
+## What is the application about (to be run with docker compose)?
+The application we will be running with docker compose has 3 main services:
+  1. `frontend` service
+  2. `backend` service
+  3. `mysql service` (database service)
+
+Also, we will be running the 4th service `phpmyadmin` which is used for managing and administering MySQL with a GUI.
+This is a very simple application for demo, where in the `mysql` database there are lists of user information and through the `frontend` service we will request to the `backend` service to get all those users and then list them in our application UI. Like this, all those application services are dependent and collaborating with each other. We will be configuring these through docker compose.
+
+## Prerequisites
+Before we start, lets make sure following things get installed into your system.
+  1. install [docker](https://www.digitalocean.com/community/tutorials/how-to-install-and-use-docker-on-ubuntu-20-04)
+  2. install [docker compose](https://www.digitalocean.com/community/tutorials/how-to-install-and-use-docker-compose-on-ubuntu-20-04)
```

Comment:
should we redirect the users to the official docker installation page ? https://docs.docker.com/engine/install/debian/#install-using-the-repository this also has a docker post installation link. I think it'll be more credible to use the official documentation

Reply:
sure